test/unit/cmd/http.test.ts

```diff
@@ -123,9 +124,10 @@ describe('http command', () => {
 				type: 'http',
 				target: 'google.com',
 				protocol: 'http',
-				query: {
+				request: {
 					method: 'get',
 					path: '400',
+					query: '',
```

Comment:
This almost seems like we don't have a single test with this property set. Also, for the `?` normalization at the beginning.

Reply:
No, I meant a test with a non-empty value, e.g. `?queryS` and also a test with `queryS` since I see you allow both a version with and without `?` at the beginning.

Reply:
How would we test that exactly? The socket message only contains the fully parsed response.

We might have to move option-parsers to another method and test them on their own.

Reply:
I see there is one test using `nock` in that file, so with that approach, it's possible. If the runner constructs the wrong URL, nock won't match it, and the test will fail.

The way other tests are written doesn't seem very good because there's no relation between the actual input params and the returned stream - the tests verify the runner can handle the output but don't verify at all what it actually asks for (what request it sends). This makes the tests very weak.

It seems the other commands have the same problem - there we should probably add asserts that the mocked executables get called with the correct params. Otherwise again, we don't verify again how they are called.

Since this is a bigger problem, I'll open a separate issue.

Reply:
added a url builder.

Reply:
+ added a nock test.